src/main/java/com/opams/arch/entities/ProductEntity.java

```diff
@@ -1,8 +1,38 @@
 package com.opams.arch.entities;
 
-import javax.persistence.Entity;
+import lombok.*;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+import java.math.BigDecimal;
 
 @Entity
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
 public class ProductEntity {
+
+    @GeneratedValue
+    @Id
+    private long productId;
+
+    @NotNull
+    @Positive
+    private BigDecimal retailPrice;
+
+    private String size;
+
+    private String color;
+
+    private String place;
+
+    private String image;   //base64 string
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    private OriginEntity origin;
```

Comment:
Is that really helpful to have only origin reference in product?
What will happen if I want to search all products in one country? Should I firstly retrieve all origins in this country and then provide that as query to filter products?